pkg/controller/pxc/vs.go

```diff
@@ -1,120 +1,96 @@
+		HaproxyVersion:    &vm.HAProxyVersion,
+		Product:           productName,
+		HTTPClient:        &http.Client{Timeout: 10 * time.Second},
 	}
+	applyParams = applyParams.WithTimeout(10 * time.Second)
 
-	req.Header.Set("Accept", "application/json")
+	resp, err := srvCl.VersionService.VersionServiceApply(applyParams)
 
-	resp, err := client.Do(req)
 	if err != nil {
 		return DepVersion{}, err
 	}
 
-	defer resp.Body.Close()
-	if resp.StatusCode != http.StatusOK {
-		return DepVersion{}, fmt.Errorf("received bad status code %s", resp.Status)
-	}
-
-	r := VersionResponse{}
-	err = json.NewDecoder(resp.Body).Decode(&r)
-	if err != nil {
-		return DepVersion{}, fmt.Errorf("failed to unmarshal response: %v", err)
-	}
-
-	if len(r.Versions) == 0 {
+	if len(resp.Payload.Versions) == 0 {
 		return DepVersion{}, fmt.Errorf("empty versions response")
 	}
 
-	pxcVersion, err := getVersion(r.Versions[0].Matrix.PXC)
+	pxcVersion, err := getVersion(resp.Payload.Versions[0].Matrix.Pxc)
 	if err != nil {
 		return DepVersion{}, err
 	}
 
-	backupVersion, err := getVersion(r.Versions[0].Matrix.Backup)
+	backupVersion, err := getVersion(resp.Payload.Versions[0].Matrix.Backup)
 	if err != nil {
 		return DepVersion{}, err
 	}
 
-	pmmVersion, err := getVersion(r.Versions[0].Matrix.PMM)
+	pmmVersion, err := getVersion(resp.Payload.Versions[0].Matrix.Pmm)
 	if err != nil {
 		return DepVersion{}, err
 	}
 
-	proxySqlVersion, err := getVersion(r.Versions[0].Matrix.ProxySQL)
+	proxySqlVersion, err := getVersion(resp.Payload.Versions[0].Matrix.Proxysql)
 	if err != nil {
 		return DepVersion{}, err
 	}
 
-	haproxyVersion, err := getVersion(r.Versions[0].Matrix.HAProxy)
+	haproxyVersion, err := getVersion(resp.Payload.Versions[0].Matrix.Haproxy)
 	if err != nil {
 		return DepVersion{}, err
 	}
 
 	return DepVersion{
-		PXCImage:        r.Versions[0].Matrix.PXC[pxcVersion].ImagePath,
+		PXCImage:        resp.Payload.Versions[0].Matrix.Pxc[pxcVersion].ImagePath,
 		PXCVersion:      pxcVersion,
-		BackupImage:     r.Versions[0].Matrix.Backup[backupVersion].ImagePath,
+		BackupImage:     resp.Payload.Versions[0].Matrix.Backup[backupVersion].ImagePath,
 		BackupVersion:   backupVersion,
-		ProxySqlImage:   r.Versions[0].Matrix.ProxySQL[proxySqlVersion].ImagePath,
+		ProxySqlImage:   resp.Payload.Versions[0].Matrix.Proxysql[proxySqlVersion].ImagePath,
 		ProxySqlVersion: proxySqlVersion,
-		HAProxyImage:    r.Versions[0].Matrix.HAProxy[haproxyVersion].ImagePath,
-		HAProxyVersion:  haproxyVersion,
-		PMMImage:        r.Versions[0].Matrix.PMM[pmmVersion].ImagePath,
+		PMMImage:        resp.Payload.Versions[0].Matrix.Pmm[pmmVersion].ImagePath,
 		PMMVersion:      pmmVersion,
+		HAProxyImage:    resp.Payload.Versions[0].Matrix.Haproxy[haproxyVersion].ImagePath,
+		HAProxyVersion:  haproxyVersion,
 	}, nil
 }
 
-func getVersion(versions map[string]Version) (string, error) {
+func getVersion(versions map[string]models.VersionVersion) (string, error) {
 	if len(versions) != 1 {
```

Comment:
if you checked length of "versions" here, you can just return first element without loop